v2/security_group_rule.go

```diff
@@ -56,18 +58,33 @@ func securityGroupRuleFromAPI(r *oapi.SecurityGroupRule) *SecurityGroupRule {
 		}(),
 		Protocol: (*string)(r.Protocol),
 		SecurityGroupID: func() (v *string) {
-			if r.SecurityGroup != nil {
+			if r.SecurityGroup != nil &&
+				(r.SecurityGroup.Visibility == nil ||
+					*r.SecurityGroup.Visibility != "public") {
 				v = r.SecurityGroup.Id
 			}
 			return
 		}(),
+		SecurityGroupName: func() (v *string) {
+			if r.SecurityGroup != nil && r.SecurityGroup.Visibility != nil &&
+				*r.SecurityGroup.Visibility == "public" {
```

Comment:
Although API might not return empty `SecurityGroup.Name`, it is still a pointer and adding ` && SecurityGroup.Name != nil` would not hurt.